lib/railway_ipc/persistence.ex

```diff
@@ -0,0 +1,66 @@
+defmodule RailwayIpc.Persistence do
+  @behaviour RailwayIpc.PersistenceBehaviour
+  @repo Application.get_env(:railway_ipc, :repo, RailwayIpc.Dev.Repo)
+  alias RailwayIpc.Persistence.PublishedMessage
+  alias RailwayIpc.Persistence.ConsumedMessage
+  alias RailwayIpc.MessageConsumption
+  alias RailwayIpc.Persistence.ConsumedMessageAdapter
+  alias RailwayIpc.Persistence.PublishedMessageAdapter
+  import Ecto.Query
+
+  def insert_published_message(message, exchange) do
```

Comment:
I ended up putting the adapter calls inside these persistence functions which I don't love but the alternative was creating yet another module that does persistence logic and calls out to this persistence module, so that we can keep the publishing/consume contexts relatively clean. any ideas @StevenNunez ?